scripts/us_census/pep/annual_population/preprocess_test.py

```diff
@@ -0,0 +1,110 @@
+    It will generate and test CSV, MCF and TMCF files for given test input files
+    and comapre it with expected results.
+    """
+
+    def __init__(self, methodName: str = ...) -> None:
+        super().__init__(methodName)
+
+        test_files = []
+        for dir_path in INPUT_DIRS:
+            files_dir = os.path.join(_MODULE_DIR, TEST_DATA_DIR, "datasets",
+                                     dir_path)
+            test_files += [
+                os.path.join(files_dir, file)
+                for file in sorted(os.listdir(files_dir))
+            ]
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            cleaned_csv_file_path = os.path.join(tmp_dir,
+                                                 "test_output_data.csv")
+            mcf_file_path = os.path.join(tmp_dir, "test_census.mcf")
+            tmcf_file_path = os.path.join(tmp_dir, "test_census.tmcf")
+
+            process(test_files, cleaned_csv_file_path, mcf_file_path,
+                    tmcf_file_path)
+
+            with open(mcf_file_path, encoding="UTF-8") as mcf_file:
+                self._actual_mcf_data = mcf_file.read()
+
+            with open(tmcf_file_path, encoding="UTF-8") as tmcf_file:
+                self._actual_tmcf_data = tmcf_file.read()
+
+            with open(cleaned_csv_file_path, encoding="utf-8") as csv_file:
+                self._actual_csv_data = csv_file.read()
+
+    def test_mcf_tmcf_files(self):
+        """
+        This method tests MCF, tMCF files generated using process module against
+        expected results.
+        """
+        expected_mcf_file_path = os.path.join(_MODULE_DIR, TEST_DATA_DIR,
+                                              "expected_files",
+                                              "usa_annual_population.mcf")
+
+        expected_tmcf_file_path = os.path.join(_MODULE_DIR, TEST_DATA_DIR,
+                                               "expected_files",
+                                               "usa_annual_population.tmcf")
+
+        with open(expected_mcf_file_path,
+                  encoding="UTF-8") as expected_mcf_file:
+            expected_mcf_data = expected_mcf_file.read()
+
+        with open(expected_tmcf_file_path,
+                  encoding="UTF-8") as expected_tmcf_file:
+            expected_tmcf_data = expected_tmcf_file.read()
+
+        self.assertEqual(expected_mcf_data.strip(),
+                         self._actual_mcf_data.strip())
+        self.assertEqual(expected_tmcf_data.strip(),
+                         self._actual_tmcf_data.strip())
+
+    def test_create_csv(self):
+        """
+        This method tests CSV file generated using process module against
+        expected CSV result.
+        """
+        expected_csv_file_path = os.path.join(_MODULE_DIR, TEST_DATA_DIR,
+                                              "expected_files",
+                                              "usa_annual_population.csv")
```

Comment:
The test output file is too large. Can you pls have a sample of row sin the input? A few 10s of rows per input file is sufficient for tests that captures various formats.

If you need to have large files (>1000 lines), pls add them as GitLFS files.

Reply:
Created sample of rows from total dataset for test data.